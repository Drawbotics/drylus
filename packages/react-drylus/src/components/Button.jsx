import React from 'react';
import { css, cx } from 'emotion';
import PropTypes from 'prop-types';
import sv from '@drawbotics/style-vars';

import { Categories, Sizes } from '../base';
import Icon from './Icon';


const styles = {
  root: css`
    background: ${sv.neutralLight};
    color: ${sv.colorPrimary};
    border-radius: ${sv.defaultBorderRadius};
    padding: calc(${sv.paddingExtraSmall} * 1.5) ${sv.defaultPadding};
    outline: 0;
    border: 0;
    transition: ${sv.defaultTransition};
    display: inline-flex;
    align-items: center;
    justify-content: center;

    &:hover {
      cursor: pointer;
      background: ${sv.neutral};
    }

    &:active {
      box-shadow: ${sv.insetActiveMedium};
    }

    &:disabled {
      cursor: not-allowed;
      background: ${sv.neutralLight} !important;
      color: ${sv.colorDisabled} !important;
      box-shadow: none;
    }
  `,
  brand: css`
    background: ${sv.brand};
    color: ${sv.white};

    &:hover {
      background: ${sv.brandDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }
  `,
  danger: css`
    background: ${sv.red};
    color: ${sv.white};

    &:hover {
      background: ${sv.redDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }
  `,
  info: css`
    background: ${sv.blue};
    color: ${sv.white};

    &:hover {
      background: ${sv.blueDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }
  `,
  success: css`
    background: ${sv.green};
    color: ${sv.white};

    &:hover {
      background: ${sv.greenDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }
  `,
  warning: css`
    background: ${sv.orange};
    color: ${sv.white};

    &:hover {
      background: ${sv.orangeDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }
  `,
  small: css`
    padding: ${sv.paddingExtraSmall} ${sv.paddingExtraSmall};
    font-size: 0.9rem;
  `,
  large: css`
    padding: ${sv.paddingSmall} ${sv.paddingHuge};
  `,
  secondary: css`
    background: transparent;
    color: ${sv.colorPrimary};
    box-shadow: 0 0 0 1px ${sv.neutral} inset;

    &:hover {
      background: ${sv.neutralLight};
    }

    &:active {
      box-shadow: ${sv.insetActiveLight};
    }
  `,
  tertiary: css`
    background: transparent;
    color: ${sv.blue};

    &:hover {
      background: ${sv.neutralLight};
      color: ${sv.blueDark};
    }

    &:active {
      box-shadow: ${sv.insetActiveLight};
    }
  `,
  rightIcon: css`
    > i {
      margin-top: -2px;
      margin-left: 4px;
    }
  `,
  leftIcon: css`
    > i {
      margin-top: -2px;
      margin-right: 4px;
    }
  `,
  iconOnly: css`
    border-radius: 1000px;
    height: ${sv.marginExtraLarge};
    width: ${sv.marginExtraLarge};
    padding: 0;

    > i {
      display: flex;
      align-items: center;
      justify-content: center;
      margin-right: 0;
      margin-left: 0;
      margin-bottom: -3px;
    }
  `,
  smallIcon: css`
    height: ${sv.marginLarge};
    width: ${sv.marginLarge};

    > i {
      font-size: 1rem;
      margin-bottom: -3px;
    }
  `,
};


export const ButtonTiers = {
  PRIMARY: 'PRIMARY',
  SECONDARY: 'SECONDARY',
  TERTIARY: 'TERTIARY',
};


const Button = ({
  children,
  disabled,
  onClick,
  category,
  size,
  tier=ButtonTiers.PRIMARY,
  icon,
  iconSide='left',
}) => {
  const iconOnly = ! children && icon;
  return (
    <button
      onClick={onClick}
      className={cx(styles.root, {
        [styles[size?.toLowerCase()]]: size,
        [styles[tier?.toLowerCase()]]: tier,
        [styles.rightIcon]: iconSide === 'right' && icon,
        [styles.leftIcon]: iconSide === 'left' && icon,
        [styles.iconOnly]: iconOnly,
        [styles.smallIcon]: iconOnly && size === 'SMALL',
        [styles[category?.toLowerCase()]]: category && tier === ButtonTiers.PRIMARY,
      })}
      disabled={disabled}>
      {do{
        if (iconSide === 'left' && icon) {
          <Icon name={icon} />
        }
      }}
      {children}
      {do{
        if (iconSide === 'right' && icon) {
          <Icon name={icon} />
        }
      }}
    </button>
  );
}


Button.propTypes = {
  /** Just text for the button */
  children: PropTypes.string,

  /** Disables the button click */
  disabled: PropTypes.bool,

  /** Triggered after the button is clicked */
  onClick: PropTypes.func,

  /** Category of the button. Can be danger, success, info, warning */
  category: PropTypes.oneOf([Categories.BRAND, Categories.DANGER, Categories.SUCCESS, Categories.INFO, Categories.WARNING]),

  /** Size of the button. Can be small, large, default */
  size: PropTypes.oneOf([Sizes.SMALL, Sizes.DEFAULT, Sizes.LARGE]),

  /** Tier of the button */
  tier: PropTypes.oneOf([ButtonTiers.PRIMARY, ButtonTiers.SECONDARY, ButtonTiers.TERTIARY]),

  /** Name of the icon to be displayed within the button. Shown on the left by default */
  icon: PropTypes.string,

  /** Side on which the icon is displayed */
  iconSide: PropTypes.oneOf(['left', 'right']),
};


Button.defaultProps = {
  iconSide: 'left',
  size: Sizes.DEFAULT,
  tier: ButtonTiers.PRIMARY,
};


export default Button;
