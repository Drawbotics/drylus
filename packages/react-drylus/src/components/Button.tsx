import sv, { fade } from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import React from 'react';

import { Category, Color, Size, Tier } from '../enums';
import { OnClickCallback, Responsive, Style } from '../types';
import { colorEnumToCategory, getEnumAsClass, run, useResponsiveProps } from '../utils';

const styles = {
  root: css`
    background: ${sv.neutralLight};
    color: ${sv.colorPrimary};
    border-radius: ${sv.defaultBorderRadius};
    padding: calc(${sv.paddingExtraSmall} * 1.5) ${sv.defaultPadding};
    outline: 0;
    border: 0;
    transition: ${sv.transitionShort};
    display: inline-flex;
    align-items: center;
    justify-content: center;

    &:hover {
      cursor: pointer;
      background: ${sv.neutral};
    }

    &:active {
      box-shadow: ${sv.insetActiveMedium};
    }

    &:disabled {
      cursor: not-allowed;
      background: ${sv.neutralLight};
      color: ${sv.colorDisabled};
      box-shadow: none;
    }
  `,
  brand: css`
    background: ${sv.brand};
    color: ${sv.white} !important;

    &:hover {
      background: ${sv.brandDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }

    &:disabled {
      background: ${fade(sv.brand, 40)};
    }
  `,
  danger: css`
    background: ${sv.red};
    color: ${sv.white} !important;

    &:hover {
      background: ${sv.redDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }

    &:disabled {
      background: ${fade(sv.red, 40)};
    }
  `,
  info: css`
    background: ${sv.blue};
    color: ${sv.white} !important;

    &:hover {
      background: ${sv.blueDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }

    &:disabled {
      background: ${fade(sv.blue, 40)};
    }
  `,
  success: css`
    background: ${sv.green};
    color: ${sv.white} !important;

    &:hover {
      background: ${sv.greenDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }

    &:disabled {
      background: ${fade(sv.green, 40)};
    }
  `,
  warning: css`
    background: ${sv.orange};
    color: ${sv.white} !important;

    &:hover {
      background: ${sv.orangeDark};
    }

    &:active {
      box-shadow: ${sv.insetActive};
    }

    &:disabled {
      background: ${fade(sv.orange, 40)};
    }
  `,
  brandAlt: css`
    color: ${sv.brand};

    &:hover {
      color: ${sv.brandDark};
    }

    &:disabled {
      color: ${fade(sv.brand, 40)};
    }
  `,
  dangerAlt: css`
    color: ${sv.red};

    &:hover {
      color: ${sv.redDark};
    }

    &:disabled {
      color: ${fade(sv.red, 40)};
    }
  `,
  warningAlt: css`
    color: ${sv.orange};

    &:hover {
      color: ${sv.orangeDark};
    }

    &:disabled {
      color: ${fade(sv.orange, 40)};
    }
  `,
  infoAlt: css`
    color: ${sv.blue};

    &:hover {
      color: ${sv.blueDark};
    }

    &:disabled {
      color: ${fade(sv.blue, 40)};
    }
  `,
  successAlt: css`
    color: ${sv.green};

    &:hover {
      color: ${sv.greenDark};
    }

    &:disabled {
      color: ${fade(sv.green, 40)};
    }
  `,
  primaryAlt: css`
    color: ${sv.colorPrimary};

    &:hover {
      color: ${sv.colorPrimary};
    }

    &:disabled {
      color: ${sv.colorDisabled};
    }
  `,
  small: css`
    padding: calc(${sv.paddingExtraSmall} - 1px) ${sv.paddingExtraSmall};
    /* font-size: 0.9rem; */
  `,
  large: css`
    padding: ${sv.paddingSmall} ${sv.paddingHuge};
  `,
  secondary: css`
    background: transparent;
    color: ${sv.colorPrimary};
    box-shadow: 0 0 0 1px ${sv.neutral} inset;

    &:hover {
      background: ${sv.neutralLight};
    }

    &:active {
      box-shadow: 0 0 0 1px ${sv.neutral} inset, ${sv.insetActiveLight};
    }

    &:disabled {
      box-shadow: 0 0 0 1px ${sv.neutralLight} inset;
      background: none;
    }
  `,
  tertiary: css`
    background: transparent;
    color: ${sv.blue};

    &:hover {
      background: ${fade(sv.neutral, 50)};
      color: ${sv.blueDark};
    }

    &:active {
      box-shadow: ${sv.insetActiveLight};
    }

    &:disabled {
      background: none;
      color: ${sv.colorDisabled};
    }
  `,
  round: css`
    border-radius: 1000px;
    height: ${sv.marginExtraLarge};
    width: ${sv.marginExtraLarge};
    padding: 0;

    i {
      display: flex;
      align-items: center;
      justify-content: center;
      margin-right: 0;
      margin-left: -1px;
      margin-bottom: -1px;
    }
  `,
  roundSmall: css`
    height: ${sv.marginLarge};
    width: ${sv.marginLarge};

    i {
      font-size: 1rem;
      margin-bottom: -1px;
      margin-left: -1px;
    }
  `,
  fullWidth: css`
    width: 100%;
    flex: 1;
  `,
  leading: css`
    margin-right: ${sv.marginExtraSmall};
    height: 0;
    display: flex;
    align-items: center;
  `,
  trailing: css`
    margin-left: ${sv.marginExtraSmall};
    height: 0;
    display: flex;
    align-items: center;
  `,
};

export const buttonStyles = styles;

export interface ButtonProps {
  children?: React.ReactNode;

  /** Disables the button click */
  disabled?: boolean;

  /** Triggered after the button is clicked */
  onClick?: OnClickCallback<HTMLElement>;

  category?: Exclude<Category, Category.PRIMARY>;

  color?: Exclude<Color, Color.PRIMARY>;

  /** @default Size.DEFAULT */
  size?: Extract<Size, Size.SMALL | Size.DEFAULT | Size.LARGE>;

  /** @default Tier.PRIMARY */
  tier?: Tier;

  /** Shown in front of the button text, can be a Spinner or Icon */
  leading?: React.ReactNode;

  /** Shown after the button text, can be a Spinner or Icon */
  trailing?: React.ReactNode;

  /** Makes button take the full width of the container */
  fullWidth?: boolean;

  /** Used for style overrides */
  style?: Style;

  /** Reponsive prop overrides */
  responsive?: Responsive<this>;
}

export const Button = ({ responsive, ...rest }: ButtonProps) => {
  const {
    children,
    disabled,
    onClick,
    category: _category,
    size = Size.DEFAULT,
    tier = Tier.PRIMARY,
    leading,
    trailing,
    fullWidth,
    style,
    color,
  } = useResponsiveProps<ButtonProps>(rest, responsive);

  if (!children && trailing && leading) {
    throw new Error('If no children are given, only pass trailing or leading, but not both');
  }

  const round = children == null && (trailing != null || leading != null);

  const category = color ? colorEnumToCategory(color) : _category;

  return (
    <button
      style={style}
      onClick={onClick}
      className={cx(styles.root, {
        [styles[getEnumAsClass<typeof styles>(size)]]: size != null,
        [styles.round]: round === true,
        [styles.roundSmall]: round === true && size === Size.SMALL,
        [styles[getEnumAsClass<typeof styles>(category)]]:
          category != null && tier === Tier.PRIMARY,
        [styles[getEnumAsClass<typeof styles>(tier)]]: tier != null,
        [styles[`${category?.toLowerCase() ?? ''}Alt` as keyof typeof styles]]:
          category != null && tier !== Tier.PRIMARY,
        [styles.fullWidth]: fullWidth === true,
      })}
      disabled={disabled}>
      {run(() => {
        if (leading) {
          return <div className={cx({ [styles.leading]: !round })}>{leading}</div>;
        }
      })}
      {children}
      {run(() => {
        if (trailing) {
          return <div className={cx({ [styles.trailing]: !round })}>{trailing}</div>;
        }
      })}
    </button>
  );
};
