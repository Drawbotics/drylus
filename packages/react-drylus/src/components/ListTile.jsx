import React from 'react';
import { css, cx } from 'emotion';
import PropTypes from 'prop-types';
import sv from '@drawbotics/style-vars';

import Flex, { FlexItem } from '../layout/Flex';


const styles = {
  root: css`
    display: inline-block;
  `,
  clickable: css`
    &:hover {
      cursor: pointer;
    }
  `,
  leading: css`
    display: flex;
    align-items: center;
    margin-right: ${sv.marginExtraSmall};
  `,
  trailing: css`
    display: flex;
    align-items: center;
  `,
  title: css`
    color: ${sv.colorPrimary};
  `,
  subtitle: css`
    color: ${sv.colorSecondary};
    font-size: 0.9rem;
  `,
  withMargin: css`
    margin-bottom: calc(${sv.marginExtraSmall} / 2);
  `,
};


const ListTile = ({
  title,
  subtitle,
  leading,
  trailing,
  onClick,
}) => {
  return (
    <div className={cx(styles.root, { [styles.clickable]: !! onClick })} onClick={onClick}>
      <Flex>
        {do {
          if (leading) {
            <FlexItem>
              <div className={styles.leading}>
                {leading}
              </div>
            </FlexItem>
          }
        }}
        <FlexItem flex>
          {do {
            if (title != null) {
              <div className={cx(styles.title, { [styles.withMargin]: subtitle })}>
                {title}
              </div>
            }
          }}
          {do {
            if (subtitle != null) {
              <div className={styles.subtitle}>
                {subtitle}
              </div>
            }
          }}
        </FlexItem>
        {do {
          if (trailing != null) {
            <FlexItem>
              <div className={styles.trailing}>
                {trailing}
              </div>
            </FlexItem>
          }
        }}
      </Flex>
    </div>
  );
};


ListTile.propTypes = {
  /** Main text to be displayed */
  title: PropTypes.string,

  /** Smaller text displayed below the title */
  subtitle: PropTypes.string,

  /** Can be anything, will appear centered to the left of the title and/or subtitle */
  leading: PropTypes.node,

  /** Can be anything, will appear centered to the right of the title and/or subtitle */
  trailing: PropTypes.node,

  /** Triggered when the component is clicked */
  onClick: PropTypes.func,
};


export default ListTile;
