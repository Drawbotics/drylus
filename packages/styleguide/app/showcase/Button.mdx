import { Button } from '@drawbotics/react-drylus';

import Playground from '../components/Playground';


# Hello

The following is an non-editable code example
```jsx
<Button type="danger">Hello</Button>
```

The following is the rendered component
<Playground component={Button}>
  <Button>Hello</Button>
  <Button>Fuck</Button>
</Playground>
