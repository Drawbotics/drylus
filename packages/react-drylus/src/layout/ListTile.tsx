import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import React from 'react';

import { Text } from '../components';
import { Responsive, Style } from '../types';
import { run, useResponsiveProps } from '../utils';
import { Flex, FlexItem } from './Flex';

const styles = {
  root: css`
    display: inline-block;
  `,
  clickable: css`
    &:hover {
      cursor: pointer;
    }
  `,
  leading: css`
    display: flex;
    align-items: center;
    margin-right: ${sv.marginExtraSmall};
  `,
  trailing: css`
    display: flex;
    align-items: center;
    margin-left: ${sv.marginExtraSmall};
  `,
  title: css`
    color: ${sv.colorPrimary};
  `,
  subtitle: css`
    color: ${sv.colorSecondary};
    font-size: 0.9rem;
  `,
  withMargin: css`
    margin-bottom: calc(${sv.marginExtraSmall} / 2);
  `,
};

interface ListTileProps {
  /** Main text to be displayed */
  title: React.ReactNode;

  /** Smaller text displayed below the title */
  subtitle?: React.ReactNode;

  /** Can be anything, will appear centered to the left of the title and/or subtitle */
  leading?: React.ReactNode;

  /** Can be anything, will appear centered to the right of the title and/or subtitle */
  trailing?: React.ReactNode;

  /** Triggered when the component is clicked */
  onClick?: () => void;

  /** Used for style overrides */
  style?: Style;

  /** Reponsive prop overrides */
  responsive?: Responsive<this>;
}

export const ListTile = ({ responsive, ...rest }: ListTileProps) => {
  const { title, subtitle, leading, trailing, onClick, style } = useResponsiveProps<ListTileProps>(
    rest,
    responsive,
  );

  return (
    <div
      style={style}
      className={cx(styles.root, { [styles.clickable]: !!onClick })}
      onClick={onClick}>
      <Flex>
        {run(() => {
          if (leading) {
            return (
              <FlexItem>
                <div className={styles.leading}>{leading}</div>
              </FlexItem>
            );
          }
        })}
        <FlexItem flex>
          {run(() => {
            if (title != null) {
              if (typeof title === 'string' || (title as React.ReactElement).type === Text) {
                return (
                  <div className={cx(styles.title, { [styles.withMargin]: subtitle != null })}>
                    {title}
                  </div>
                );
              } else {
                return <div className={styles.withMargin}>{title}</div>;
              }
            }
          })}
          {run(() => {
            if (subtitle != null) {
              if (typeof subtitle === 'string' || (subtitle as React.ReactElement).type === Text) {
                return <div className={styles.subtitle}>{subtitle}</div>;
              } else {
                return subtitle;
              }
            }
          })}
        </FlexItem>
        {run(() => {
          if (trailing != null) {
            return (
              <FlexItem>
                <div className={styles.trailing}>{trailing}</div>
              </FlexItem>
            );
          }
        })}
      </Flex>
    </div>
  );
};
