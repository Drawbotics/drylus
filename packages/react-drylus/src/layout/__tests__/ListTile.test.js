import React from 'react';
import { create } from 'react-test-renderer';

import ListTile from '../ListTile';
import Button from '../../components/Button';


describe('ListTile', () => {
  describe('matches snapshot when', () => {
    it('has a title', () => {
      const tree = create(
        <ListTile title="Title" />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a subtitle', () => {
      const tree = create(
        <ListTile title="Title" subtitle="Subtitle" />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a custom title', () => {
      const tree = create(
        <ListTile title={<Button>Button</Button>} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a trailing and leading component', () => {
      const tree = create(
        <ListTile
          title="Title"
          trailing={<Button>Trailing</Button>}
          leading={<Button>Leading</Button>} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  
  describe('triggers when clicked', () => {
    let triggered = false;

    expect(triggered).toBeFalsy();

    const component = create(
      <ListTile title="Title" onClick={() => triggered = true} />
    );

    component.root.props.onClick();
    
    expect(triggered).toBeTruthy();
  });
});