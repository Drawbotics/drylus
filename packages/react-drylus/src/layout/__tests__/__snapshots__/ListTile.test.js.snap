// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListTile matches snapshot when has a custom title 1`] = `
.css-1kgtu2h-Flex__root {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
}

.css-8312sn-ListTile__root {
  display: inline-block;
}

.css-94tip-ListTile__withMargin {
  margin-bottom: calc(8px / 2);
}

.css-qq01sn-Button__root {
  background: #eaeff4;
  color: #172b4e;
  border-radius: 5px;
  padding: calc(8px * 1.5) 24px;
  outline: 0;
  border: 0;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.css-qq01sn-Button__root:hover {
  cursor: pointer;
  background: #d3dce6;
}

.css-qq01sn-Button__root:active {
  box-shadow: 0 1px 6px #879ba6 inset;
}

.css-qq01sn-Button__root:disabled {
  cursor: not-allowed;
  background: #eaeff4;
  color: rgba(135,155,166,0.5);
  box-shadow: none;
}

.css-iempa9-Flex__item-Flex__equalSpan {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="css-8312sn-ListTile__root"
>
  <div
    className="css-1kgtu2h-Flex__root undefined"
  >
    <div
      className="css-iempa9-Flex__item-Flex__equalSpan"
      style={Object {}}
    >
      <div
        className="css-94tip-ListTile__withMargin"
      >
        <button
          className="css-qq01sn-Button__root"
        >
          Button
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ListTile matches snapshot when has a subtitle 1`] = `
.css-1kgtu2h-Flex__root {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
}

.css-8312sn-ListTile__root {
  display: inline-block;
}

.css-1wodq5h-ListTile__subtitle {
  color: #879ba6;
  font-size: 0.9rem;
}

.css-iempa9-Flex__item-Flex__equalSpan {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.css-rhrm0e-ListTile__title-ListTile__withMargin {
  color: #172b4e;
  margin-bottom: calc(8px / 2);
}

<div
  className="css-8312sn-ListTile__root"
>
  <div
    className="css-1kgtu2h-Flex__root undefined"
  >
    <div
      className="css-iempa9-Flex__item-Flex__equalSpan"
      style={Object {}}
    >
      <div
        className="css-rhrm0e-ListTile__title-ListTile__withMargin"
      >
        Title
      </div>
      <div
        className="css-1wodq5h-ListTile__subtitle"
      >
        Subtitle
      </div>
    </div>
  </div>
</div>
`;

exports[`ListTile matches snapshot when has a title 1`] = `
.css-1kgtu2h-Flex__root {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
}

.css-8312sn-ListTile__root {
  display: inline-block;
}

.css-12l3x5o-ListTile__title {
  color: #172b4e;
}

.css-iempa9-Flex__item-Flex__equalSpan {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="css-8312sn-ListTile__root"
>
  <div
    className="css-1kgtu2h-Flex__root undefined"
  >
    <div
      className="css-iempa9-Flex__item-Flex__equalSpan"
      style={Object {}}
    >
      <div
        className="css-12l3x5o-ListTile__title"
      >
        Title
      </div>
    </div>
  </div>
</div>
`;

exports[`ListTile matches snapshot when has a trailing and leading component 1`] = `
.css-1kgtu2h-Flex__root {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
}

.css-8312sn-ListTile__root {
  display: inline-block;
}

.css-1nr0qeo-ListTile__leading {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 8px;
}

.css-1vo6ywr-ListTile__trailing {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: 8px;
}

.css-12l3x5o-ListTile__title {
  color: #172b4e;
}

.css-qq01sn-Button__root {
  background: #eaeff4;
  color: #172b4e;
  border-radius: 5px;
  padding: calc(8px * 1.5) 24px;
  outline: 0;
  border: 0;
  -webkit-transition: all 0.15s ease-in-out;
  transition: all 0.15s ease-in-out;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.css-qq01sn-Button__root:hover {
  cursor: pointer;
  background: #d3dce6;
}

.css-qq01sn-Button__root:active {
  box-shadow: 0 1px 6px #879ba6 inset;
}

.css-qq01sn-Button__root:disabled {
  cursor: not-allowed;
  background: #eaeff4;
  color: rgba(135,155,166,0.5);
  box-shadow: none;
}

.css-iempa9-Flex__item-Flex__equalSpan {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<div
  className="css-8312sn-ListTile__root"
>
  <div
    className="css-1kgtu2h-Flex__root undefined"
  >
    <div
      className="css-19bskwp-Flex__item"
      style={Object {}}
    >
      <div
        className="css-1nr0qeo-ListTile__leading"
      >
        <button
          className="css-qq01sn-Button__root"
        >
          Leading
        </button>
      </div>
    </div>
    <div
      className="css-iempa9-Flex__item-Flex__equalSpan"
      style={Object {}}
    >
      <div
        className="css-12l3x5o-ListTile__title"
      >
        Title
      </div>
    </div>
    <div
      className="css-19bskwp-Flex__item"
      style={Object {}}
    >
      <div
        className="css-1vo6ywr-ListTile__trailing"
      >
        <button
          className="css-qq01sn-Button__root"
        >
          Trailing
        </button>
      </div>
    </div>
  </div>
</div>
`;
